src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+            originalDetector.getCustomResultIndexOrAlias(),
+            originalDetector.getImputationOption(),
+            originalDetector.getRecencyEmphasis(),
+            originalDetector.getSeasonIntervals(),
+            newHistoryIntervals,
+            originalDetector.getRules(),
+            originalDetector.getCustomResultIndexMinSize(),
+            originalDetector.getCustomResultIndexMinAge(),
+            originalDetector.getCustomResultIndexTTL(),
+            originalDetector.getFlattenResultIndexMapping(),
+            null,
+            originalDetector.getFrequency(),
+            originalDetector.getAutoCreated()
+        );
+    }
+
+    // Use AD suggest API to find better interval/window-delay/history intervals based on the historical data
+    private <T> void suggestHyperParametersPhase(AnomalyDetector detector, String tenantId, int maxRetries, ActionListener<T> listener) {
+        log.info("Starting suggest api step");
+
+        // Create suggest request for interval, history, window_delay
+        SuggestConfigParamRequest suggestRequest = new SuggestConfigParamRequest(
+            AnalysisType.AD,
+            detector,
+            "interval,history,window_delay",
+            TimeValue.timeValueSeconds(SUGGEST_API_TIMEOUT_SECONDS)
+        );
+
+        adClient.suggestAnomalyDetector(suggestRequest, new ActionListener<SuggestConfigParamResponse>() {
+            @Override
+            public void onResponse(SuggestConfigParamResponse response) {
+                try {
+                    AnomalyDetector optimizedDetector = applySuggestionsToDetector(detector, response);
+                    validateModelPhase(optimizedDetector, tenantId, maxRetries, 0, listener);
+
+                } catch (Exception e) {
+                    validateModelPhase(detector, tenantId, maxRetries, 0, listener);
+                }
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // Continue to model validation with original detector even if suggest fails
+                validateModelPhase(detector, tenantId, maxRetries, 0, listener);
+            }
+        });
+    }
+
+    // Checks if there's enough data to train the model (final validation before creating detector)
+    private <T> void validateModelPhase(
+        AnomalyDetector detector,
+        String tenantId,
+        int maxRetries,
+        int currentRetry,
+        ActionListener<T> listener
+    ) {
+        log.info("Starting model validation");
+
+        callValidationAPI(detector, "model", new ActionListener<ValidateConfigResponse>() {
+            @Override
+            public void onResponse(ValidateConfigResponse response) {
+
+                if (response.getIssue() != null) {
+                    String issueAspect = response.getIssue().getAspect().toString();
+                    boolean isBlockingError = issueAspect != null && issueAspect.toLowerCase(Locale.ROOT).startsWith("detector");
```

Comment:
Is starting with "detector" always blocking?

Is there some helper method somewhere which defines blocking errors that you can use?